index.d.ts

```diff
@@ -303,24 +347,38 @@ declare class Emittery {
 		});
 	```
 	*/
-	anyEvent(): AsyncIterableIterator<unknown>
+	anyEvent(): AsyncIterableIterator<
+		[
+			EventNameFromDataMap<EventDataMap>,
+			EventDataMap[EventNameFromDataMap<EventDataMap>]
+		]
+	>;
 
 	/**
 	Remove an `onAny` subscription.
 	*/
-	offAny(listener: (eventName: EventName, eventData?: unknown) => void): void;
+	offAny(
+		listener: (
+			eventName: EventNameFromDataMap<EventDataMap> | EmptyEvents,
+			eventData?: EventDataMap[EventNameFromDataMap<EventDataMap>]
+		) => void
+	): void;
 
 	/**
 	Clear all event listeners on the instance.
 
 	If `eventName` is given, only the listeners for that event are cleared.
 	*/
-	clearListeners(eventName?: EventNames): void;
+	clearListeners(
+		eventName?: EventNameFromDataMap<EventDataMap> | EmptyEvents
```

Comment:
`EventNameFromDataMap<EventDataMap> | EmptyEvents` is repeated a lot. Maybe it should be a type alias.

Reply:
Refactored to make all the types simpler!